### ISSUE TITLE
generate skip function automatically

### ISSUE THREAD

**Issue by sils1297** [coala-analyzer/coala#1129]
we could generate the skip_test

like `skip_test = generate_skip_test(MarkdownBear.BINARY)`

or with https://github.com/coala-analyzer/coala/issues/1072 we could use that automatically ideally...

**Comment by sils1297** [comment]
Ok, now we have the `check_prerequisites`, we could easily generate a skip for bears using that, simply instantiate the bear and if runtime error was raised grab the log message as skip reason.

@Makman2 can you do this? This should be quick for you and very useful.

**Comment by Makman2** [comment]
I think we should make `check_prerequisites` a staticmethod, so this would be possible then:
```
skip_test = MarkdownBear.check_prerequisites
```
together with a function that allows to easily check for existence of a binary (that does the `popen` stuff etc).

**Comment by sils1297** [comment]
@Makman2 hey didn't you do this a few days ago? :/

**Comment by Makman2** [comment]
@sils1297 for the `CorrectionBasedBear` at least, for bears using `Lint` it's not working yet since `check_prerequisites` is not adequately implemented^^

**Comment by Makman2** [comment]
I think after merging `CorrectionBasedBear` and `Lint` this will be solved automatically^^

**Comment by Makman2** [comment]
See also #1225.
